packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/hub/FlutterHubElement.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_datastore.types.hub
+
+import com.amazonaws.amplify.amplify_datastore.types.model.FlutterSerializedModel
+import com.amplifyframework.core.model.temporal.Temporal
+import com.amplifyframework.datastore.appsync.ModelMetadata
+import com.amplifyframework.datastore.appsync.ModelWithMetadata
+import com.amplifyframework.datastore.appsync.SerializedModel
+import com.amplifyframework.datastore.syncengine.LastSyncMetadata
+
+class FlutterHubElement(
+        private val modelWithMetadata: ModelWithMetadata<*>
+) {
+    fun toValueMap(): Map<String, Any> {
+        var serializedModel: MutableMap<String, Any> = FlutterSerializedModel(this.modelWithMetadata.model as SerializedModel).toMap().toMutableMap()
```

Comment:
Is this `this.modelWithMetadata.model as SerializedModel)` guaranteed to succeed everytime? In my observe API case, I was getting events for system models as well which would fail here.

Reply:
@richardmcclellan Do you have any insight on @Amplifiyer's question?  It's not clear to me from the library code whether other model types can be fired off in Hub events.

Reply:
Nothing _guarantees_ it, at compile time anyway.    However, if it was a system model, it would be a bug.   Changes to system models are definitely not supposed to be exposed in Hub events, since those are for internal use only.  So, it should always be a `SerializedModel`, but I would suggest just validating that it is a `SerializedModel` before you create the `FlutterHubElement` object, and if it's not, drop the event, and log a warning or something.

Reply:
@richardmcclellan ok thanks - i'll implement this.

Reply:
Done.